adobe/spectrum-web-components
# fix(overlay): ensure that passing "open" to the directive manages a single strategy

## Description
Ensure that the Trigger Directive calculates the `type` of a strategy so that it can be compared over time and not spuriously create a new strategy and subsequently new `<sp-overlay>` elements.

## How has this been tested?
    1. Go [here](https://managed-overlay-directive--spectrum-web-components.netlify.app/storybook/index.html?path=/story/overlay-directive--managed-overlay-trigger)
    2. Click the "Create Overlay Render Button And Open Overlay" button
    3. See that the "Toggle Popover" button is appended and that an Overlay is opened.
    4. Close the Overlay
    5. Click "Toggle Popover"
    6. See that the Overlay opens without cycling and without appending multiple Overlay elements to the page.\

## Types of changes

## Checklist
